### PR TITLE
93 week 5 view hazards filters UI

### PR DESCRIPTION
**Pull request by BryanCaldeira**
Filter UI - (Functionality will be implemented in another card)
- `CheckBox.js` common component (new).
- `ModalFilter.js` common component (new).
- Map.js default coordinated changed to focus on BC, Canada (comment).
- Clear search results on click of back button (comment).
- Display only markers based on quick filters (comment).
- Icons color changed to default and applied using CSS.
- `flyTo()` function created for all flyTo calls in home screen.
- Map.createLayerGroups() has been modified due to a bug that was not letting me to remove the layers. This update won't break any past calls to the function.
- Quick filters had issues and was not applying based on location, If user had changed his/her location the filter was still applying based on current location. Hence `positionSecondary` will be used to keep track of location change on `user search` and quick filters are working as expected now.
- The params name passed to `/hazard-report` endpoint for `longitude` was misspelt correction has been made `lon -> lng`. Now results based on current location are displaying as expected.

Trello cards
- [93-week-5-view-hazards-filters-ui](https://trello.com/c/7ob5IpOU/93-week-5-view-hazards-filters-ui)
- [135-ui-update-home-screen](https://trello.com/c/4ABYXBRV/135-ui-update-home-screen)
- [133-fix-the-zoom-level-of-the-home-page](https://trello.com/c/tTvfHRMm/133-fix-the-zoom-level-of-the-home-page)

### CODE REVIEW

File: pages/home/script.js [hunk 1]
```diff
@@ -326,21 +307,24 @@ const onSearchInput = debounce(async ({ target }) => {
...
-          target: '.sb-suggestion-wrapper',
-        };
-      }) ?? []
-    );
-
-    suggestionOnClick();
-
-    boxSuggestion.style.display = 'block';
-    boxCategories.style.display = 'none';
+  if(!searchSuggestions.length) {
```
Comment (BryanCaldeira) [inline pages/home/script.js:310]: search suggestion box-shadow showing up was fixed here

File: pages/home/style.css [hunk 2]
```diff
@@ -11,15 +11,17 @@ body {
   overflow: hidden;
 }
 
-/* header {
+header {
```
Comment (BryanCaldeira) [inline pages/home/style.css:14]: header position needed for home screen as rest of the positioning is absolute

File: assets/models/Map.js [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-  mapLayers = {};
+  mapLayers = new L.LayerGroup();
   currentMarker = null;
   static CURRENT_ZOOM = 12;
   static MAP_ID = 'map';
   static MAX_ZOOM = 22;
   static DEFAULT_MAP_ZOOM = 12; // If we don't set the zoom level, 12 is the default of Leaflet
   static DEFAULT_LOCATION = {
-    lat: 49.2,
-    lng: -123.12,
```
Comment (BryanCaldeira) [inline assets/models/Map.js:12]: Focus is now on BC, Canada, Previously it was on Vancouver, BC, Canada

File: assets/css/searchbar.css [hunk 4]
```diff
@@ -79,8 +85,19 @@
   padding-inline: 16px;
 }
 
-.sb-suggestion-icon {
+.sb-suggestion-item-wrapper .sb-suggestion-icon {
   margin-right: 10px;
+  background-color: var(--neutral-500);
```
Comment (BryanCaldeira) [inline assets/css/searchbar.css:90]: Cut short addresses to not overflow

File: assets/css/searchbar.css [hunk 5]
```diff
@@ -79,8 +85,19 @@
...
+}
+
+.sb-suggestion-txt--box {
+  overflow: hidden;
+}
+
+.sb-suggestion-txt--box p {
+  overflow: hidden;
+  white-space: nowrap;
+  text-overflow: ellipsis;
```
Comment (BryanCaldeira) [inline assets/css/searchbar.css:100]: Cut short addresses to not overflow within the box

File: assets/css/quick-filter.css [hunk 6]
```diff
@@ -24,18 +24,8 @@
...
-  background: var(--primary-900, #0f1815) !important;
-  border-color: var(--primary-900, #0f1815) !important;
-}
-
-.quick-filter.selected .quick-filter__name {
-  color: var(--white) !important;
-}
-
 .quick-filter.selected .quick-filter__icon {
-  background-color: var(--white) !important;
```
Comment (BryanCaldeira) [inline assets/css/quick-filter.css:38]: !important property was not required. Moved these styles after `:hover` selector

File: assets/css/searchbar.css [hunk 7]
```diff
@@ -69,7 +75,7 @@
   border-radius: 12px;
   border-radius: 12px;
   background-color: var(--white);
-  max-height: 148px;
+  max-height: 500px;
```
Comment (BryanCaldeira) [inline assets/css/searchbar.css:78]: displays at-most 5 search result cards

File: pages/home/script.js [hunk 8]
```diff
@@ -36,15 +36,12 @@ let position =
     ? { lat: latitude, lng: longitude }
     : Map.DEFAULT_LOCATION;
 
-// let positionObj = {};
-// let locationDetails = '';
+let positionSecondary = {};
```
Comment (BryanCaldeira) [inline pages/home/script.js:39]: keep track of position when user changes location based on search

File: pages/home/script.js [hunk 9]
```diff
@@ -36,15 +36,12 @@ let position =
...
 let hazardCardParams = {};
 let searchSuggestions = [];
 let reports = [];
 let categoryFilters = [];
 let flyToTrigger = true;
-const FLY_TO_ZOOM = 12;
-const ANIMATION_DURATION = 4;
```
Comment (BryanCaldeira) [inline pages/home/script.js:47]: these variables were not required as `flyTo` is a new function being used to fly to any location

File: assets/components/ModalFilter.js [hunk 10]
```diff
@@ -1,28 +1,43 @@
...
+      <div class="modal-filter--wrapper-outer">
+        <div class="modal-filter--wrapper-inner">
+          ${ categories?.map(item => {
+            return `
+              <div class="modal-filter--catergories-wrapper-outer">
+                <h3>${ item?.name }</h3>
+                <div class="modal-filter--categories-wrapper-inner">
+                  ${ 
+                    item?.options?.map(option => {
+                      return  CheckBox(option?.name?.toLowerCase(), option?.name);
```
Comment (whamester) [inline assets/components/ModalFilter.js:25]: I think you might want to send the option id to the checkbox in the first argument here.

**Pull request merged**